Add feature: zeros_like() from a dense tensor to a sparse tensor

vinnamkim opened the pull request:
Fixes #67904.
 - Create a sparse tensor when the sparse layout is given even if the input tensor is not sparse.

cc @nikitaved @pearu @cpuhrsch @IvanYashchuk

Commit:
Fix zeros_like
 - Create a sparse tensor when the sparse layout is given even if the input tensor is not sparse.

Signed-off-by: Vinnam Kim <vinnam.kim@makinarocks.ai>

Commit:
Remove test code
 - Remove test code that expects an exception that will no longer be thrown by the previous commit

Signed-off-by: Vinnam Kim <vinnam.kim@makinarocks.ai>

Commit:
Add test
 - Compare shapes between the dense input and the sparse output

Signed-off-by: Vinnam Kim <vinnam.kim@makinarocks.ai>

IvanYashchuk reviewed (approved):
Thank you, @vinnamkim, for your contribution!
@cpuhrsch, could you please take a look?

Commit:
Add more tests

 - Add more tests to check shapes and layout
 - Fix shape test assertTrue -> assertEqual

Signed-off-by: Vinnam Kim <vinnam.kim@makinarocks.ai>

cpuhrsch reviewed (approved):
LGTM

vinnamkim commented on test/test_sparse.py:
```diff
@@ -1833,11 +1833,9 @@ def test_shape(i_shapes, v_shapes, nnzs):
-            result = torch.zeros_like(dense_tensor, layout=torch.sparse_coo)
+        dense_tensor = sparse_tensor.to_dense()
+        result = torch.zeros_like(dense_tensor, layout=torch.sparse_coo)
+        self.assertTrue(dense_tensor.shape, result.shape)
```
Added those tests you suggest in https://github.com/pytorch/pytorch/pull/68108/commits/2f593e1514269650f79c18cf2a2ebd25db2e0d66. Thank you.
